### DISCUSSION TITLE
When RUN java/tests/vread_vwrite/run_test.pl, I have a problem

### DISCUSSION THREAD

**Discussion by rla0h** [OpenDDS/OpenDDS#4559] [Q&A] [answered]
I want to JSON encode/decode practice using example, but That example's ./run_test.pl is not working.
Output is ERROR: decode_from_string failed 2

I found that RETCODE 2 is unsupported, then is 3.27.0 version not apply now?

I'm sorry that i'm not good at English.
Please help me.. Thank you
![image](https://github.com/OpenDDS/OpenDDS/assets/88080716/19d5e21c-df9f-4468-9a39-4dd90a8ed269)

**Comment by jrw972** [comment, marked as answer]
You probably didn't compile with support for json.

If you used the configure script, you need to add `--rapidjson`.
Note that running configure again with new parameters may or may not work.  If it doesn't, just start with a clean repo or release archive.

@iguessthislldo What do we need to do for rapidjson with CMake?  Does that look like:
1. Populate the rapidjson submodule.
2. Define `OPENDDS_RAPIDJSON` to the directory containing rapidjson during the cmake configure step

**Reply by iguessthislldo** [reply]
RapidJSON is enabled and downloaded by default in both the configure script and CMake. In the configure scripts case it could silently fail unless `--rapidjson` is passed.

**Reply by iguessthislldo** [reply]
https://github.com/OpenDDS/OpenDDS/blob/19bd1cbadbe2e3760d79e37d1c60fa3e0c66c4b5/CMakeLists.txt#L50-L60

**Reply by rla0h** [reply]
Thank you for your help!!!!
God bless you!
